### ISSUE TITLE
JavacCompiler fails to parse warnings, ignoring all output

### ISSUE THREAD

**Issue by rfscholte** [codehaus-plexus/plexus-compiler#37]
The following output results in no messages at all:

```
warning: [path] bad path element "d:\maven_repo\.m2\repository\org\ow2\asm\asm-xml\5.0.3\asm-5.0.3.jar": no such file or directory
warning: [path] bad path element "d:\maven_repo\.m2\repository\org\ow2\asm\asm-xml\5.0.3\asm-util-5.0.3.jar": no such file or directory
warning: [options] bootstrap class path not set in conjunction with -source 1.7
3 warnings
An exception has occurred in the compiler (9). Please file a bug against the Java compiler via the Java bug reporting page (http://bugreport.java.com) after checking the Bug Database (http://bugs.java.com) for duplicates. Include your program and the following diagnostic in your report. Thank you.
java.lang.NullPointerException
	at jdk.zipfs/jdk.nio.zipfs.JarFileSystem.getVersionMap(JarFileSystem.java:137)
	at jdk.zipfs/jdk.nio.zipfs.JarFileSystem.createVersionedLinks(JarFileSystem.java:112)
	at jdk.zipfs/jdk.nio.zipfs.JarFileSystem.<init>(JarFileSystem.java:85)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:134)
	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager$ArchiveContainer.<init>(JavacFileManager.java:517)
	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.getContainer(JavacFileManager.java:319)
	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.list(JavacFileManager.java:715)
	at jdk.compiler/com.sun.tools.javac.code.ClassFinder.list(ClassFinder.java:722)
	at jdk.compiler/com.sun.tools.javac.code.ClassFinder.scanUserPaths(ClassFinder.java:655)
	at jdk.compiler/com.sun.tools.javac.code.ClassFinder.fillIn(ClassFinder.java:526)
	at jdk.compiler/com.sun.tools.javac.code.ClassFinder.complete(ClassFinder.java:293)
	at jdk.compiler/com.sun.tools.javac.code.Symbol.complete(Symbol.java:633)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$PackageSymbol.members(Symbol.java:1120)
	at jdk.compiler/com.sun.tools.javac.code.Symtab.listPackageModules(Symtab.java:810)
	at jdk.compiler/com.sun.tools.javac.comp.Enter.visitTopLevel(Enter.java:344)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:529)
	at jdk.compiler/com.sun.tools.javac.comp.Enter.classEnter(Enter.java:285)
	at jdk.compiler/com.sun.tools.javac.comp.Enter.classEnter(Enter.java:300)
	at jdk.compiler/com.sun.tools.javac.comp.Enter.complete(Enter.java:570)
	at jdk.compiler/com.sun.tools.javac.comp.Enter.main(Enter.java:554)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.enterTrees(JavaCompiler.java:1052)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:923)
	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:302)
	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:162)
	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
```

**Comment by adamvoss** [comment]
I've been working to debug why, when I set `fork` to true with the Maven compiler plugin, not all messages will be logged.  Specifically I have found that warning either are not returned or will only be partially returned.  I have traced it as far as `JavacCompile.performCompile` returning with missing or clipped messages.

This is particularly problematic when one enables `failOnWarning` compilation fails no feedback at all as to why it failed.

Have I correctly identified this as the issue tracking what I have encountered?

**Comment by rfscholte** [comment]
According to https://github.com/codehaus-plexus/plexus-compiler/blob/master/plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java#L710 it should be able to collect warnings. It shouldn't be too hard to write a unittest passing the original output to this method.

**Comment by adamvoss** [comment]
@rfscholte That line and a unit test over this case was added by #40.  Is there a nightly build available or do you have an idea when the next release may be?

**Comment by rfscholte** [comment]
I've done a deployment to https://oss.sonatype.org/content/repositories/plexus-snapshots/
It is probably up to @olamy or me to do a release.

**Comment by adamvoss** [comment]
I feel I must be missing an obvious mistake.  Shouldn't the following POM cause it to use the snapshot dependency?

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0  http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>group</groupId>
   <artifactId>artifact</artifactId>
   <version>1.0</version>
   <pluginRepositories>
      <pluginRepository>
         <id>Sonatype Plexus Snapshots</id>
         <url>https://oss.sonatype.org/content/repositories/plexus-snapshots/</url>
         <snapshots>
            <enabled>true</enabled>
         </snapshots>
         <releases>
            <enabled>true</enabled>
         </releases>
      </pluginRepository>
   </pluginRepositories>
   <build>
      <plugins>
         <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-compiler-plugin</artifactId>
            <version>3.7.0</version>
            <dependencies>
               <dependency>
                  <groupId>org.codehaus.plexus</groupId>
                  <artifactId>plexus-compiler-javac</artifactId>
                  <version>2.8.3-SNAPSHOT</version>
               </dependency>
            </dependencies>
         </plugin>
      </plugins>
   </build>
</project>
```

Even when I set the version to something that isn't a snapshot, it still won't seem to resolve to anything other than the original version for compiler plugin when I use `mvn dependency:resolve-plugins`

**Comment by adamvoss** [comment]
Found [MNG-5588](https://issues.apache.org/jira/browse/MNG-5588).  Looks like it is not possible to use the snapshot version without rebuilding the plugin.

**Comment by rfscholte** [comment]
I've released 2.8.3
